src/components/Annotation.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  useContext,
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+import { createPortal } from 'react-dom';
+import MapContext from '../context/MapContext';
+import AnnotationProps from './AnnotationProps';
+
+export default function Annotation({
+  latitude,
+  longitude,
+
+  title = '',
+  subtitle = '',
+  accessibilityLabel = null,
+
+  selected = undefined,
+  onSelect = undefined,
+  onDeselect = undefined,
+  children,
+}: AnnotationProps) {
+  const [annotation, setAnnotation] = useState<mapkit.Annotation | null>(null);
+  const contentRef = useRef<HTMLDivElement>(document.createElement('div'));
+  const map = useContext(MapContext);
+
+  // Coordinates
+  useEffect(() => {
+    if (map === null) return undefined;
+    if (!contentRef.current) return undefined;
```

Comment:
This check isn’t necessary since `contentRef.current` always contains an object, you can remove it.

```suggestion
    if (map === null) return undefined;
```